scipy/stats/_continuous_distns.py

```diff
@@ -7943,27 +7944,117 @@ class argus_gen(rv_continuous):
+                           op_flags=[['readonly']])
+            while not it.finished:
+                idx = tuple((it.multi_index[j] if not bc[j] else slice(None))
+                            for j in range(-len(self._size), 0))
+                r = self._rvs_scalar(it[0], numsamples)
+                out[idx] = r.reshape(shp)
+                it.iternext()
+
+        if self._size == ():
+            out = out[()]
+        return out
+
+    def _rvs_scalar(self, chi, numsamples=None):
+        # if chi <= 2.611:
+        # use rejection method, see Devroye:
+        # Non-Uniform Random Variate Generation, 1986, section II.3.2.
+        # write: self.pdf = c * g(x) * h(x), where
+        # h is [0,1]-valued and g is a density
+        # g(x) = d1 * chi**2 * x * exp(-chi**2 * (1 - x**2) / 2), 0 <= x <= 1
+        # h(x) = sqrt(1 - x**2), 0 <= x <= 1
+        # Integrating g, we get:
+        # G(x) = d1 * exp(-chi**2 * (1 - x**2) / 2) - d2
+        # d1 and d2 are determined by G(0) = 0 and G(1) = 1
+        # d1 = 1 / (1 - exp(-0.5 * chi**2))
+        # d2 = 1 / (exp(0.5 * chi**2) - 1)
+        # => G(x) = (exp(chi**2 * x**2 /2) - 1) / (exp(chi**2 / 2) - 1)
+        # expected number of iterations is c with
+        # c = -np.expm1(-0.5 * chi**2) * chi / (_norm_pdf_C * _argus_phi(chi))
+        # note that G can be inverted easily, so we can sample
+        # rvs from this distribution
+        # G_inv(y) = sqrt(2 * log(1 + (exp(chi**2 / 2) - 1) * y) / chi**2)
+        # to avoid an overflow of exp(chi**2 / 2), it is convenient to write
+        # G_inv(y) = sqrt(1 + 2 * log(exp(-chi**2 / 2) * (1-y) + y) / chi**2)
+        #
+        # if chi > 2.611:
+        # use ratio of uniforms method applied to a transformed variable of X
+        # (X is ARGUS with parameter chi):
+        # Y = chi * sqrt(1 - X**2) has density proportional to
+        # u**2 * exp(-u**2 / 2) on [0, chi] (Maxwell distribution conditioned
+        # on [0, chi]). Apply ratio of uniforms to this density to generate
+        # samples of Y and convert back to X
+        # the expected number of iterations is below 2.2 for all chi > 0.
+        # the value 2.611 can be found by comparing the rejection constants
+        # of the ratio of uniforms method and the rejection method.
+        # it can be shown that one should use the rejection method for chi
+        # less than 2.611 and the ratio of uniforms method for larger chi.
+        # at chi ~ 2.611, the rejection constants of both methods coincide
+
+        if chi <= 2.611:
+            # use rejection method
+            size1d = tuple(np.atleast_1d(numsamples))
+            N = int(np.prod(size1d))
+            x = np.zeros(N)
+            echi = np.exp(-chi**2 / 2)
+            simulated = 0
+            while simulated < N:
+                k = N - simulated
+                u = self._random_state.uniform(size=k)
+                v = self._random_state.uniform(size=k)
```

Comment:
These might need to change if there is a decision to move forward with PR: 
https://github.com/scipy/scipy/pull/11394
I haven't reviewed that PR and there are a couple linked ones that Warren has posted. I'm not requesting these two lines change now but want to call it out to prevent break on merge, if #11394 goes in before this one.

Reply:
The ordering of this and gh-11394 should not be a problem.  (Add `size` and `random_state` as arguments to `_rvs` and `_rvs_scalar` and use instead of `self._size`, `self._random_state`. And `_rvs_scalar` already has a `num_samples` argument.)
So a small change no matter the order.